wellcomeml/io/epmc/client.py

```diff
@@ -0,0 +1,302 @@
+                    )
+                    return None
+                else:
+                    self.logger.error(
+                        "epmc.get_full_text: http-%d pmid=%s",
+                        e.response.status_code,
+                        pmid
+                    )
+            raise
+        return None
+
+    def get_references(self, session, pub_id, source='MED'):
+        """
+        Fetches references from EPMC's REST API for a given publication pub_id and source
+        (e.g. 'MED')
+        """
+        params = {"format": "json", "page": 1, "pageSize": 1000}
+
+        epmc_references_url = "/".join(
+            [self.api_endpoint, source, str(pub_id), 'references']
+        )
+
+        go_to_next_page = True
+        references = []
+
+        while go_to_next_page:
+            try:
+                response = session.get(
+                    epmc_references_url,
+                    params=params
+                )
+                response_json = response.json()
+                current_page_result = response_json.get('referenceList', {}).get('reference', [])
+                if not current_page_result:
+                    go_to_next_page = False
+
+                references += current_page_result
+                params['page'] += 1
+            except requests.RequestException:
+                self.logger.error(
+                    "epmc.get_references: requests error pub_id=%s query=%s", pub_id, params
+                )
+                go_to_next_page = False
+
+        if references:
+            return references
+        else:
+            return None
+
+    def get_citations(self, session, pub_id, source='MED'):
+        """
+        Fetches citations from EPMC's REST API for a given publication pub_id and source
+        (e.g. 'MED')
+        """
+
+        params = {"format": "json", "page": 1, "pageSize": 1000}
+
+        epmc_references_url = "/".join(
+            [self.api_endpoint, source, str(pub_id), 'citations']
+        )
+
+        go_to_next_page = True
+        citations = []
+
+        while go_to_next_page:
+            try:
+                response = session.get(
+                    epmc_references_url,
+                    params=params
+                )
+                response_json = response.json()
+                current_page_result = response_json.get('citationList', {}).get('citation', [])
+                if not current_page_result:
+                    go_to_next_page = False
+
+                citations += current_page_result
+                params['page'] += 1
+            except requests.RequestException:
+                self.logger.error(
+                    "epmc.get_citations: requests error pub_id=%s query=%s", pub_id, params
+                )
+                go_to_next_page = False
+
+        if citations:
+            return citations
+        else:
+            return None
+
+
+if __name__ == "__main__":
```

Comment:
I'm not sure if this script belongs here. Maybe as an 'example' inside `wellcomeml/examples`, or maybe as part of a CLI, as you mentioned? The way it stands, people that only install the library from pip won't be able to use `__main__`.

Reply:
It does not even work I think. I will put something in examples